GameTracker-Core/DirectorySearchHelper.cs

```diff
@@ -0,0 +1,83 @@
+﻿using GameTracker_Core.Models;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace GameTracker_Core
+{
+    public static class DirectorySearchHelper
+    {
+        public static bool DirectoryExecuteSearch(string dir)
+        {
+            return DirectoryExecuteSearchRecursive(dir, false);
+        }
+
+        private static bool DirectoryExecuteSearchRecursive(string dir, bool erg)
+        {
+            try
+            {
+                foreach (string f in Directory.GetFiles(dir))
+                {
+                    if (Path.GetExtension(f).Equals(".exe"))
+                    {
+                        //Console.WriteLine(Path.GetFileName(f));
+                        return true;
+                    }
+                }
+                foreach (string d in Directory.GetDirectories(dir))
+                {
+                    //Console.WriteLine(Path.GetFileName(d));
+                    erg = DirectoryExecuteSearchRecursive(d, erg);
```

Comment:
You should break the loop as soon as `erg` is `true` as it could be overriden with `false` in the next iteration